ts/src/otelAppInsights.ts

```diff
@@ -0,0 +1,62 @@
+import { app, LogLevel } from '@azure/functions';
+import { AzureMonitorLogExporter, AzureMonitorTraceExporter } from '@azure/monitor-opentelemetry-exporter';
+import { context as otelContext, propagation } from '@opentelemetry/api';
+import { SeverityNumber } from '@opentelemetry/api-logs';
+import { getNodeAutoInstrumentations, getResourceDetectors } from '@opentelemetry/auto-instrumentations-node';
+import { registerInstrumentations } from '@opentelemetry/instrumentation';
+import { detectResourcesSync } from '@opentelemetry/resources';
+import { LoggerProvider, SimpleLogRecordProcessor } from '@opentelemetry/sdk-logs';
+import { NodeTracerProvider, SimpleSpanProcessor } from '@opentelemetry/sdk-trace-node';
+
+const resource = detectResourcesSync({ detectors: getResourceDetectors() });
+
+const tracerProvider = new NodeTracerProvider({ resource });
+tracerProvider.addSpanProcessor(new SimpleSpanProcessor(new AzureMonitorTraceExporter()));
+tracerProvider.register();
+
+const loggerProvider = new LoggerProvider({ resource });
+loggerProvider.addLogRecordProcessor(new SimpleLogRecordProcessor(new AzureMonitorLogExporter()));
+
+registerInstrumentations({ tracerProvider, loggerProvider, instrumentations: [getNodeAutoInstrumentations()] });
+
+// NOTE: The below code will soon be a part of a new package `@opentelemetry/instrumentation-azure-functions`
+// See here for more info: https://github.com/Azure/azure-functions-nodejs-library/issues/245
+app.setup({ capabilities: { WorkerOpenTelemetryEnabled: true } });
+
+const logger = loggerProvider.getLogger('default');
+app.hook.log((context) => {
+    logger.emit({
+        body: context.message,
+        severityNumber: toOtelSeverityNumber(context.level),
+        severityText: context.level,
+    });
+});
+
+app.hook.preInvocation((context) => {
+    context.functionHandler = otelContext.bind(
+        propagation.extract(otelContext.active(), {
+            traceparent: context.invocationContext.traceContext.traceParent,
```

Comment:
Just checking that these keys are meant to be `traceparent` and `tracestate` and not `traceParent` and `traceState`?

Reply:
Annoyingly yes, this is correct. If the casing matched, I could just pass `context.invocationContext.traceContext` directly